### PR TITLE
fix(connector): adyen cancel

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Bugfix

## Description

Fixed adyen cancel.

## Motivation and Context

Adyen cancel was broken because of contract changes. Fixed it.

## How did you test it?

image

## Checklist

- [x] I formatted the code `cargo +nightly fmt`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/adyen/transformers.rs [hunk 1]
```diff
@@ -417,7 +415,7 @@ impl TryFrom<types::PaymentsCancelResponseRouterData<AdyenCancelResponse>>
     fn try_from(
         item: types::PaymentsCancelResponseRouterData<AdyenCancelResponse>,
     ) -> Result<Self, Self::Error> {
-        let status = match item.response.response.as_str() {
+        let status = match item.response.status.as_str() {
             "received" => storage_enums::AttemptStatus::Voided,
             "processing" => storage_enums::AttemptStatus::Pending,
             _ => storage_enums::AttemptStatus::VoidFailed,
```
Comment (Narayanbhat166) [inline crates/router/src/connector/adyen/transformers.rs:421]: This can go into a from implementation.
response_status -> adyen_status -> attempt_status

**Review by Narayanbhat166 — CHANGES_REQUESTED** [review]
address the transformation changes

**Review by Narayanbhat166 — DISMISSED** [review]
LGTM :)

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

**Review by SanchithHegde — APPROVED** [review]

File: crates/router/src/connector/adyen/transformers.rs [hunk 2]
```diff
@@ -173,18 +173,23 @@ pub struct AdyenCard {
...
+    status: CancelStatus,
 }
 
+#[derive(Default, Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum CancelStatus {
+    Received,
+    #[default]
+    Processing,
+}
```
[the remark this answers is not in the document]
Comment (dracarys18) [reply crates/router/src/connector/adyen/transformers.rs:192]: Noted I believe that has changes only to payment. Adyen sends different status for cancel

[the remark this answers is not in the document]
Comment (SanchithHegde) [reply crates/router/src/connector/adyen/transformers.rs:192]: I'm aware of the statuses being different for cancel, we even got it to compile on her system.
If your PR gets merged first (which is very likely), I'll suggest her to ignore changes to the cancel statuses.

[the remark this answers is not in the document]
Comment (dracarys18) [reply crates/router/src/connector/adyen/transformers.rs:192]: Cool

**Review by jarnura — APPROVED** [review]

**Pull request merged**